invenio_communities/communities/services/components.py

```diff
@@ -70,10 +70,18 @@ class CommunityAccessComponent(ServiceComponent):
     def _populate_access_and_validate(self, identity, data, record, **kwargs):
         """Populate and validate the community's access field."""
         if record is not None and "access" in data:
-            # populate the record's access field with the data already
-            # validated by marshmallow
+            access = data.get("access", {})
+            new_record_access = access.get("visibility")
```

Comment:
```suggestion
            access = data.get("access", {}).get("visibility")
```

Reply:
I am using the access variable further down, but can modify this part if we decide to raise an exception, as suggested later.